tests/ci/merge_pr.py

```diff
@@ -0,0 +1,198 @@
+            # available as a header 'last-modified' of /{org}/{repo}/commits/{sha}.
+            # Unfortunately, it's formatted as 'Wed, 04 Jan 2023 11:05:13 GMT'
+
+            commit = self.pr.head.repo.get_commit(self.pr.head.sha)
+            if commit.stats.last_modified is None:
+                logging.warning(
+                    "Unable to get info about the commit %s", self.pr.head.sha
+                )
+                return False
+
+            last_changed = datetime.strptime(
+                commit.stats.last_modified, "%a, %d %b %Y %H:%M:%S GMT"
+            )
+            if self.approved_at < last_changed:
+                logging.info(
+                    "There are changes after approve at %s",
+                    self.approved_at.isoformat(),
+                )
+                return False
+            return True
+
+        logging.info("The PR #%s is not approved", self.pr.number)
+        return False
+
+
+def parse_args() -> argparse.Namespace:
+    pr_info = PRInfo()
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Script to download binary artifacts from S3. Downloaded artifacts "
+        "are renamed to clickhouse-{static_binary_name}",
+    )
+    parser.add_argument(
+        "--check-approved",
+        action="store_true",
+        help="if set, checks that the PR is approved and no changes required",
+    )
+    parser.add_argument("--check-green", default=True, help=argparse.SUPPRESS)
+    parser.add_argument(
+        "--no-check-green",
+        dest="check_green",
+        action="store_false",
+        default=argparse.SUPPRESS,
+        help="(dangerous) if set, skip check commit to having all green statuses",
+    )
+    parser.add_argument(
+        "--repo",
+        default=pr_info.repo_full_name,
+        help="PR number to check",
+    )
+    parser.add_argument(
+        "--pr",
+        type=int,
+        default=pr_info.number,
+        help="PR number to check",
+    )
+    parser.add_argument(
+        "--token",
+        type=str,
+        default="",
+        help="a token to use for GitHub API requests, will be received from SSM "
+        "if empty",
+    )
+    args = parser.parse_args()
+    args.pr_info = pr_info
+    return args
+
+
+def main():
+    logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
+    args = parse_args()
+    logging.info("Going to process PR #%s in repo %s", args.pr, args.repo)
+    token = args.token or get_best_robot_token()
+    gh = GitHub(token, per_page=100)
+    repo = gh.get_repo(args.repo)
+    pr = repo.get_pull(args.pr)
+    if pr.is_merged():
+        logging.info("The PR #%s is already merged", pr.number)
+        return
+
+    not_ready_to_merge = pr.draft or "WIP" in pr.title
+    if not_ready_to_merge:
+        logging.info("The PR #%s is not ready for merge, stopping", pr.number)
+        return
+
+    if args.check_green:
```

Comment:
It should also check that the "Run Check" exists; otherwise, it's possible to have a few green checks and no other checks for a short period of time after the change.

Reply:
For simplicity, you can use "there are > 100 checks" as the condition.

Reply:
`do not test`?

Reply:
Actually, both suggestions are rather bad.

We don't have `Run Check` in backports. Because it's set in CheckLabels. And both backports and docs PRs have much fewer checks than 100. `Mergeable Check` is the one that can be used :thinking: 

> otherwise, it's possible to have a few green checks and no other checks for a short period of time after the change.

I don't get your concern. The script is supposed to be launched at the very last step. Sure, it's a bad idea to launch it every step. What do you mean particularly?

Reply:
> The script is supposed to be launched at the very last step.

Didn't know this. Then everything is ok.

Reply:
Ok. Just to clarify, it's executed on the FinishCheck of [pull requests](https://github.com/ClickHouse/ClickHouse/blob/45c24dd91507b7032ba1d390c4fe33d8c29791c5/.github/workflows/pull_request.yml#L4391), [backport PRs](https://github.com/ClickHouse/ClickHouse/blob/45c24dd91507b7032ba1d390c4fe33d8c29791c5/.github/workflows/backport_branches.yml#L686) and [docs PRs](https://github.com/ClickHouse/ClickHouse/blob/45c24dd91507b7032ba1d390c4fe33d8c29791c5/.github/workflows/docs_check.yml#L172)